octree_web_viewer/src/bin/points_web_viewer.rs

```diff
@@ -12,47 +12,43 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+extern crate actix;
+extern crate actix_web;
 extern crate byteorder;
-extern crate cgmath;
 #[macro_use]
 extern crate clap;
-extern crate iron;
-extern crate json;
 extern crate octree_web_viewer;
 extern crate point_viewer;
-extern crate router;
-extern crate time;
-extern crate urlencoded;
 
-use iron::mime::Mime;
-use iron::prelude::*;
+use actix_web::http::Method;
+use actix_web::{server, HttpRequest, HttpResponse};
 use octree_web_viewer::backend::{NodesData, VisibleNodes};
 use point_viewer::octree;
-use router::Router;
 use std::path::PathBuf;
-use std::sync::{Arc, RwLock};
+use std::sync::Arc;
 
 const INDEX_HTML: &'static str = include_str!("../../client/index.html");
 const APP_BUNDLE: &'static str = include_str!("../../../target/app_bundle.js");
 const APP_BUNDLE_MAP: &'static str = include_str!("../../../target/app_bundle.js.map");
 
-fn index(_: &mut Request) -> IronResult<Response> {
-    let content_type = "text/html".parse::<Mime>().unwrap();
-    Ok(Response::with((content_type, iron::status::Ok, INDEX_HTML)))
+const DEFAULT_PORT: &str = "5433";
+
+fn index(_req: &HttpRequest) -> HttpResponse {
+    HttpResponse::Ok()
+        .content_type("text/html")
+        .body(INDEX_HTML)
 }
 
-fn app_bundle(_: &mut Request) -> IronResult<Response> {
-    let content_type = "text/html".parse::<Mime>().unwrap();
-    Ok(Response::with((content_type, iron::status::Ok, APP_BUNDLE)))
+fn app_bundle(_req: &HttpRequest) -> HttpResponse {
+    HttpResponse::Ok()
+        .content_type("text/html")
+        .body(APP_BUNDLE)
 }
 
-fn app_bundle_source_map(_: &mut Request) -> IronResult<Response> {
-    let content_type = "text/html".parse::<Mime>().unwrap();
-    Ok(Response::with((
-        content_type,
-        iron::status::Ok,
-        APP_BUNDLE_MAP,
-    )))
+fn app_bundle_source_map(_req: &HttpRequest) -> HttpResponse {
+    HttpResponse::Ok()
+        .content_type("text/html")
+        .body(APP_BUNDLE_MAP)
 }
 
 fn main() {
```

Comment:
can you add a TODO to convert this to structopt (as we do inside our repo?).

Reply:
added on line 50